tests/UriTest.php

```diff
@@ -563,4 +563,11 @@ public function userInfoProvider(): array
             'invalid-enc' => ['%ZZ', '%GG', '%25ZZ:%25GG'],
         ];
     }
+
+    public function testToStringShouldNotModifyInternalState(): void
+    {
+        $uri = Uri::createFromString('https://github.com/thephpleague/uri');
+        $uri->__toString();
+        self::assertEquals($uri, Uri::createFromString('https://github.com/thephpleague/uri'));
```

Comment:
this assertion is wrong, two URI object are equals only if their method call returns the same value not if there internal state matches.